include/valijson/validation_results.hpp

```diff
@@ -51,6 +51,36 @@ class ValidationResults
     };
 
     /**
+     * @brief  Iterator for all validation errors.
+     *
+     * This struct is used to get a reference to an iterator for errors.
+     */
+    struct ErrorRange
+    {
+        typedef std::deque<Error>::const_iterator ErrorIterator;
+        ErrorRange(const std::deque<Error> &errors) : m_errors(errors) {};
+
+        ErrorIterator begin() {
+            return m_errors.begin();
+        }
+
+        ErrorIterator end() {
+            return m_errors.end();
+        }
+
+    private:
+        const std::deque<Error> &m_errors;
```

Comment:
Copying seems rather excessive, since we simply want to expose a chap, non-destructive way of iterating over accumulated errors. Are you concerned by the fact that pushing new errors would invalidate iterators? That's fair, but is `pushError` ever expected to be used by the client code? As for `popError`, I would expect most clients of the library to prefer range-based `for` instead of popping from the back. As per usual guarantees in the STL, modifying the underlying collection often invalidates the iterators, so I think that clients might be happy to accept that and simply not mix those two. I can't think of any use case where it would be useful to modify `ValidationResults` rather once validator has done it's job.

N.b. `ValidationResults` is really just collection of errors, so we could just add `begin` and `end` to the class itself, although it would read funny with that name:
```
for (const auto &error : validation_results) {
    // ...
}
```
I can't think of anything else this could hold, since positive validation doesn't deserve a mention. So perhaps we could rename this to `ValidationErrors`, make `pushError`/`popError` private (and perhaps befriend the validator), and expose iteration-friendly API?